In [5]:
# importowanie stats z modułu scipy - korzystasz z funkcji statystycznych dostępnych w bibliotece SciPy
from scipy import stats

# Zadanie 4 - Test dla wartości oczekiwanej dwóch prób zależnych

Dane przedstawiają emisję zanieczyszczeń powietrza pyłami polimerów (w t) pochodzącymi
z kilku zakładów szczególnie uciążliwych dla wybranego województwa:

 w roku I oraz
5 lat później, w roku II.


Rok I 220 185 270 285 200 295 255 190 225 230


Rok II 190 175 215 260 215 195 260 150 155 175


Na poziomie istotności 0,05 należy zweryfikować hipotezę, że nastąpił istotny spadek emisji
pyłów w tych zakładach.


## 1. Sformułowanie hipotez


H0: (μ_1  =μ_2): nie ma istotnej roznicy w spadku emisji pylow


H1: (μ_1>μ_2): w roku II znacznie spadla emisja pylow


## 2. Wybór testu statystycznego
- Test parametryczny - wartość przeciętna
- Podwójna próba
- Skala ilorazowa

-> Test T

## 3. Poziom istotności  
α=0,05


## 4. Obliczenie statystyki testowej i p-value oraz podjęcie decyzji weryfikacyjnej

In [6]:
# Wyniki przed badaniem
rok_i = [220, 185 ,270, 285, 200, 295, 255, 190, 225, 230]

# Wyniki po badaniu
rok_ii = [190, 175, 215, 260, 215, 195, 260, 150, 155, 175]

# Poziom istotności
alpha = 0.05

# Przeprowadzenie testu t dla dwóch powiązanych próbek
t_statistic, p_value = stats.ttest_rel(rok_i, rok_ii, alternative='greater')

# Wyświetlenie wyników testu
print("Statystyka t:", t_statistic)
print("Wartość p:", p_value)


# Sprawdzenie warunku na istotność statystyczną
if p_value < alpha:
    print("\nNa podstawie tej próby można stwierdzić, że emisja pylow w drugim roku była mniejsza niż w pierwszym.")
else:
    print("\nNie ma wystarczających dowodów na to, aby stwierdzić, że emisja pylow w drugim roku była mniejsza niż w pierwszym.")


Statystyka t: 3.286273677873095
Wartość p: 0.004717146628580168

Na podstawie tej próby można stwierdzić, że emisja pylow w drugim roku była mniejsza niż w pierwszym.


Z otrzymanego wydruku możemy odczytać m.in.  wartość statystyki testowej (t = 3.286273677873095) oraz  p-wartość = 0.004717146628580168. \
Ponieważ: p<α, zatem nie ma podstaw do odrzucenia hipotezy zerowej. 

Na podstawie tej próby można stwierdzić, że emisja pylow w drugim roku była mniejsza niż w pierwszym.


## 5. Wnioski:
Na podstawie tej próby można stwierdzić, że emisja pylow w drugim roku była mniejsza niż w pierwszym.
